rawhttp-core/src/main/java/rawhttp/core/body/BodyReader.java

```diff
@@ -113,17 +112,8 @@ public void writeDecodedTo(OutputStream out) throws IOException {
      *                                  by the {@link HttpBodyEncodingRegistry}.
      */
     public void writeDecodedTo(OutputStream out, int bufferSize) throws IOException {
-        DecodingOutputStream decodedStream = framedBody.getBodyDecoder().decoding(out);
-        try {
+        try (DecodingOutputStream decodedStream = framedBody.getBodyDecoder().decoding(out)) {
             framedBody.getBodyConsumer().consumeDataInto(asRawStream(), decodedStream, bufferSize);
-            decodedStream.close();
```

Comment:
Why did you remove this section? 
Without it, the caller thread could hang.

Reply:
its still there. i changed it to java `try-with-resources Statement`:
```
try (DecodingOutputStream decodedStream = framedBody.getBodyDecoder().decoding(out))
```
will close the `decodedStream` once it exists the "try" section

Reply:
Oh right! I strayed away from Java for too long 😅